avm/src/vm/memory.rs

```diff
@@ -72,6 +72,15 @@ impl HandlerMemory {
     }
   }
 
+  /// Because of constraints in Tokio's RWLock, this method moves the contents of `self` into the
+  /// `target`. The `self` is destroyed in the process
+  pub fn replace(self: HandlerMemory, target: &mut HandlerMemory) {
```

Comment:
hmmm, not sure how I feel about this. my intuition tells me maybe there is a better way to do this?

Reply:
I tried updating tokio which appeared to have the function I was looking for, but then it failed to compile because the latest Tokio depends on a super-recent version of Rust, so I went back and wrote this. At only 6 LOC I'm not too bothered, and the perf impact is minimal.

Reply:
(A super recent version of Rust that we aren't targeting, I mean.)

Reply:
But on that front, I am thinking about potentially changing the `io!` opcode definition as the current form with the Box-wrapped async function seems heavyweight and I wonder if we can do it better (I don't see a single `io!` opcode not following that pattern, for instance).